ecosystem/indexer-grpc/indexer-grpc-file-store/src/processor.rs

```diff
@@ -98,13 +121,31 @@ impl Processor {
                 anyhow::ensure!(metadata.chain_id == cache_chain_id, "Chain ID mismatch.");
                 metadata.version
             } else {
+                // If metadata doesn't exist, create and upload it and init file store latest version in cache.
+                while self
+                    .file_store_operator
+                    .update_file_store_metadata_with_timeout(cache_chain_id, 0)
+                    .await
+                    .is_err()
+                {
```

Comment:
@larry-aptos do we need to do this if we already upload the metadata in `new`

Reply:
ah good catch; i don't think so now.

Reply:
let me update this PR

Reply:
and we can redeploy scratchnet

Reply:
just for context: this was added since we considered metadata file is optional during startup but now we enforce this during `new` which should solve the issue.

Reply:
(this is now doable since chain id is passed in as config.)